### PR TITLE
Firebase refactor to work with Scholarship servlet(s)

### PR DESCRIPTION
**Pull request by MikolJiron**
A refactor of Firebase code that adheres to changes made in a PR that directly uses these methods. Use optionals where necessary to streamline null-checking and avoid NPEs. Update tests.

### CODE REVIEW

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
   private static final String BATCH_SIZE_NOT_SPECIFIED = "Batch size not specified, please send a batch size for query";
   private static final String EXCEPTION_DNE_FORMATTER = "Document with id: %s does not exist.";
```
Comment (CarlosM10011) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: Spell out does not exist please.

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java [hunk 2]
```diff
@@ -96,7 +95,7 @@ public static void storeMultipleDocuments(Firestore database, String collectionT
     throws FirebaseException {
     for (Object object : objects) {
       try {
-        storeDocument(database, collectionToWriteTo, object, null);
+        storeDocument(database, collectionToWriteTo, object, Optional.ofNullable(null));
```
Comment (Lorraine-Lyu) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: You can use `Optional.empty()`

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]

**Review by Lorraine-Lyu — CHANGES_REQUESTED** [review]
use optional.empty(), other things LGTM

**Review by CarlosM10011 — APPROVED** [review]

**Review by Lorraine-Lyu — APPROVED** [review]

**Review by hzimz — APPROVED** [review]

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java [hunk 3]
```diff
@@ -57,18 +57,17 @@
...
+  public static void storeDocument(Firestore database, String collectionToWriteTo, Object object, Optional<String> documentID) 
       throws FirebaseException {
     // Access the correct collection.
     CollectionReference collectionRef = database.collection(collectionToWriteTo);
 
     // Access/create the new document.
     DocumentReference documentRef;
-    Optional<String> optionalID = Optional.ofNullable(documentID);
-    if (!optionalID.isPresent()) {
+    if (documentID.isEmpty()) {
```
Comment (tinagrasso) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: Here and below, please follow Optional api. This should be documentId.isPresent().

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#isPresent--

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java (hunk 2 above)
Comment (tinagrasso) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: Please reference the Optional documentation. This should be Optional.empty().

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#empty--

**Review by tinagrasso — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java (hunk 3 above)
Comment (MikolJiron) [reply src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: Lol someone told me to change to isEmpty(), I guess this was right originally...

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java (hunk 2 above)
Comment (MikolJiron) [reply src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java]: Sg, thanks!

**Review by tinagrasso — APPROVED** [review]

**Pull request merged**